src/bin/julialauncher.rs

```diff
@@ -179,6 +183,680 @@ fn is_interactive() -> bool {
+    }) {
+        // no_argument: v, h, q, i
+        // optional_argument: O, g
+        return !matches!(short, 'v' | 'h' | 'q' | 'i' | 'O' | 'g');
+    }
+
+    // Long options: list no_argument and optional_argument (everything else requires an argument)
+    !matches!(
+        opt,
+        // no_argument options
+        "--version" | "--help" | "--help-hidden" | "--interactive" | "--quiet"
+            | "--experimental" | "--lisp" | "--image-codegen" | "--rr-detach"
+            | "--strip-metadata" | "--strip-ir" | "--gc-sweep-always-full"
+            | "--trace-compile-timing"
+            // optional_argument options
+            | "--project" | "--code-coverage" | "--track-allocation" | "--optimize"
+            | "--min-optlevel" | "--debug-info" | "--worker" | "--trim" | "--trace-eval"
+    )
+}
+
+/// Initialize the active project (Julia's init_active_project)
+/// Returns the project file path based on --project flag or JULIA_PROJECT env
+/// https://github.com/JuliaLang/julia/blob/dd80509227adbd525737244ebabc95ec5d634354/base/initdefs.jl#L263-L272
+fn init_active_project_impl(
+    args: &[String],
+    current_dir: &Path,
+    julia_project: Option<&str>,
+    depot_path: Option<&std::ffi::OsStr>,
+) -> Result<Option<PathBuf>> {
+    // Check for --project flag in args
+    // Stop parsing at "--" or the first positional argument (non-flag)
+    // to match Julia's argument parsing behavior
+    let mut project_cli: Option<Option<String>> = None;
+    let mut index = 1;
+    while index < args.len() {
+        let arg = &args[index];
+
+        // Stop at -- separator (everything after is for the script)
+        if arg == "--" {
+            break;
+        }
+
+        // Stop at first positional argument (doesn't start with -)
+        if !arg.starts_with('-') {
+            break;
+        }
+
+        if ["--project", "--projec", "--proje", "--proj"].contains(&arg.as_str()) {
+            project_cli = Some(None);
+        } else if let Some(value) = ["--project=", "--projec=", "--proje=", "--proj="]
+            .iter()
+            .find_map(|prefix| arg.strip_prefix(prefix))
+        {
+            project_cli = Some(Some(value.to_string()));
+        } else if julia_option_requires_arg(arg) {
+            // This option consumes the next token as its argument
+            // Skip it to avoid treating the argument as a flag
+            index += 1;
+        }
+        index += 1;
+    }
+
+    // Determine project spec
+    let project = if let Some(spec) = project_cli {
+        // --project flag takes precedence
+        // If --project has no value, treat as "@." (search upward)
+        Some(spec.unwrap_or_else(|| "@.".to_string()))
+    } else {
+        // Check JULIA_PROJECT env
+        julia_project.map(|v| {
+            if v.trim().is_empty() {
+                "@.".to_string() // Empty JULIA_PROJECT means "@."
+            } else {
+                v.to_string()
+            }
+        })
+    };
```

Comment:
A lot of this stuff I think we should just handle with clap.rs (e.g., via #1236), then we don't have to write a custom parser for command line options

Reply:
These need to exactly mimic julia's style of parsing. (Like the weird `--proj` stuff). But if clap can be made to do that after your PR we can switch to it.